### PR TITLE
Update document for property `engine` in `azurerm_kusto_cluster`

### PR DESCRIPTION
**Pull request by liuwuliuyun**
Update document for property `engine` in `azurerm_kusto_cluster`

### CODE REVIEW

**Comment by liuwuliuyun** [comment]
I am adding this doc for property engine in azurerm_kusto_cluster because several customers have reached out to me saying that migration from v2 to v3 through terraform has cause data loss.

**Comment by liuwuliuyun** [comment]
I also checked all other forceNew properties in `azurerm_kusto_cluster`, the rest are all good.

File: website/docs/r/kusto_cluster.html.markdown [hunk 1]
```diff
@@ -82,7 +82,9 @@ The following arguments are supported:
 
 * `zones` - (Optional) Specifies a list of Availability Zones in which this Kusto Cluster should be located. Changing this forces a new Kusto Cluster to be created.
 
-* `engine` - (Optional). The engine type that should be used. Possible values are `V2` and `V3`. Defaults to `V2`.
+* `engine` - (Optional). The engine type that should be used. Possible values are `V2` and `V3`. Defaults to `V2`. Changing this forces a new Kusto Cluster to be created.
+
+~> **NOTE:** In v4.0 and later version, default engine type will be `V3`.
```
Comment (katbyte) [inline website/docs/r/kusto_cluster.html.markdown:87]: quoting
```suggestion
~> **NOTE:** In `v4.0` and later version, default engine type will be `V3`. 
```

also what v4 are you referring to?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/kusto_cluster.html.markdown (hunk 1 above)
Comment (liuwuliuyun) [reply website/docs/r/kusto_cluster.html.markdown:87]: I am refering to the azurerm provider here, let me change the discription

File: website/docs/r/kusto_cluster.html.markdown [hunk 2]
```diff
@@ -82,7 +82,9 @@ The following arguments are supported:
 
 * `zones` - (Optional) Specifies a list of Availability Zones in which this Kusto Cluster should be located. Changing this forces a new Kusto Cluster to be created.
 
-* `engine` - (Optional). The engine type that should be used. Possible values are `V2` and `V3`. Defaults to `V2`.
+* `engine` - (Optional). The engine type that should be used. Possible values are `V2` and `V3`. Defaults to `V2`. Changing this forces a new Kusto Cluster to be created.
+
+~> **NOTE:** In v4.0 and later version of the AzureRM Provider, default engine type will be changed to `V3`.
```
Comment (katbyte) [inline website/docs/r/kusto_cluster.html.markdown:87]: quotes
```suggestion
~> **NOTE:** In `v4.0.0` and later version of the AzureRM Provider, default engine type will be changed to `V3`. 
```

and we should be making this change in code behind the 4.0 flag so its not missed and can be trial'd?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/kusto_cluster.html.markdown (hunk 2 above)
Comment (liuwuliuyun) [reply website/docs/r/kusto_cluster.html.markdown]: > 

https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/kusto/kusto_cluster_resource.go#L294-L301 the code we need has been added, only the doc needs to be updated.

**Comment by liuwuliuyun** [comment]
@katbyte change has been made :)

File: website/docs/r/kusto_cluster.html.markdown (hunk 2 above)
Comment (katbyte) [reply website/docs/r/kusto_cluster.html.markdown]: ahh perfect! thanks

**Review by katbyte — APPROVED** [review]
LGTM 🎩

**Pull request merged**